src/Common/src/CoreLib/System/Text/ValueStringBuilder.cs

```diff
@@ -21,7 +21,22 @@ public ValueStringBuilder(Span<char> initialBuffer)
             _pos = 0;
         }
 
-        public int Length => _pos;
+        public int Length
+        {
+            get => _pos;
+            set
+            {
+                int delta = value - _pos;
+                if (delta > 0)
+                {
+                    Append('\0', delta);
```

Comment:
Is this Length setter really useful? I do not see why anybody would actually want to do this given the shape of this class.

Indexer that let's you see and edit the content of the pending string would be more efficient replacement for the one place where this is used in DosMatcher.cs.

Reply:
> Indexer that let's you see and edit the content of the pending string would be more efficient replacement for the one place where this is used in DosMatcher.cs.

That's probably the case, cc: @JeremyKuhne.  For this PR I simply want to get rid of the copy of the code.  I'd like to leave any updates here for a subsequent PR.  Will open an issue.